src/inventory.h

```diff
@@ -122,20 +122,13 @@ struct ItemStack
 	// Returns true if the item is (was) a tool
 	bool addWear(s32 amount, IItemDefManager *itemdef)
 	{
-		if(getDefinition(itemdef).type == ITEM_TOOL)
-		{
-			if(amount > 65535 - wear)
-				clear();
-			else if(amount < -wear)
-				wear = 0;
-			else
-				wear += amount;
-			return true;
-		}
+		if(amount > 65535 - wear && getDefinition(itemdef).type == ITEM_TOOL)
```

Comment:
IMHO a better name would be WEAR_MAX. However, if the 65535 is replaced with a constant, even if it's U16_MAX instead of WEAR_MAX, then for consitency it should also be replaced everywhere else where 65536 is used for maximum wear. Using U16_MAX or WEAR_MAX in some places, and 65535 in others makes the code messy, and _harder_ to maintain. 

Symbolic constants are not recommended for their own sake. They are not a goal, but a means. The goal is to make code more maintainable. If making a constant symbolic decreases understandability of the code, or maintainability, then it should definitely _not_ be done.